### PR TITLE
fix: [#152] make sure using Fragment, FragmentManager and more

### PR DESCRIPTION
**Pull request by andhikayuana**
# Whats Changed?

- Change `android.app.Activity` to `androidx.fragment.app.FragmentActivity`
- Change `android.app.Fragment` to `androidx.fragment.app.Fragment`
- Change `getFragmentManager` to `getSupportFragmentManager ` from androidx
- add new logic `!app.getSupportFragmentManager().isStateSaved()` to `killAllFragments`

### CODE REVIEW

File: app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         mainLayout.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                getActivity().getFragmentManager().beginTransaction().remove(FragmentPin.this).commit();
```
Comment (kcw-grunt) [inline app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java:68]: My recollection is getActivity() is deprecated?  Otherwise what is your thinking on this @andhikayuana ?

File: app/src/main/java/com/breadwallet/tools/animation/BRAnimator.java [hunk 2]
```diff
@@ -324,10 +324,11 @@ public void run() {
         } else return false;
     }
 
-    public static void killAllFragments(Activity app) {
+    public static void killAllFragments(FragmentActivity app) {
```
Comment (kcw-grunt) [inline app/src/main/java/com/breadwallet/tools/animation/BRAnimator.java:327]: nice...I'm a weirdo when it comes to naming classes..the more the verbose the better. You will see many abbr. (😎) and they making the code really hard to follow. 
Thanks for this @andhikayuana

**Review by kcw-grunt — APPROVED** [review]
LGTM

File: app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java (hunk 1 above)
Comment (andhikayuana) [reply app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java:68]: so on android have 2 different fragment class

- [android.app.Fragment](https://developer.android.com/reference/android/app/Fragment)
- [androidx.fragment.app.Fragment](https://developer.android.com/reference/kotlin/androidx/fragment/app/package-summary)


actually `getActivity()` still there but it will return `nullable`, do you want to use it ? more [info](https://developer.android.com/reference/kotlin/androidx/fragment/app/Fragment#getActivity())

using `requireActivity()` assuming that the activity not null, if null it will be throwing an exception
https://developer.android.com/reference/kotlin/androidx/fragment/app/Fragment#requireActivity()

File: app/src/main/java/com/breadwallet/tools/animation/BRAnimator.java (hunk 2 above)
Comment (andhikayuana) [reply app/src/main/java/com/breadwallet/tools/animation/BRAnimator.java:327]: sure 🤝

File: app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java (hunk 1 above)
Comment (kcw-grunt) [reply app/src/main/java/com/breadwallet/presenter/fragments/FragmentPin.java:68]: Yes. This is fine for now @andhikayuana
